src/main/java/net/snowflake/client/jdbc/FieldMetadata.java

```diff
@@ -122,4 +129,10 @@ public List<FieldMetadata> getFields() {
   public void setFields(List<FieldMetadata> fields) {
     this.fields = fields;
   }
+
+  public static FieldMetadata withTypeName(String typeName) {
```

Comment:
I'm not sure, this looks strange. Maybe we can just have a constructor? Or have a better name like `ofTypeName`? `with` sounds like a builder.